ark/phenotyping/cell_cluster_utils.py

```diff
@@ -604,58 +604,139 @@ def cell_consensus_cluster(fovs, channels, base_dir, cell_som_cluster_cols, max_
+    meta_expr_col_avg_path = os.path.join(base_dir, cell_meta_expr_col_avg_name)
+
+    # check paths
+    io_utils.validate_paths([cell_som_input_data_path, som_expr_col_avg_path])
+
+    # if the column average file for cell meta clusters already exists, skip
+    if os.path.exists(meta_expr_col_avg_path):
+        print("Already generated average expression file for cell meta clusters, skipping")
+        return
+
     # compute the average pixel SOM/meta counts per cell meta cluster
-    print("Compute the average number of pixel SOM/meta cluster counts per cell meta cluster")
-    cell_meta_cluster_avgs_and_counts = compute_cell_cluster_count_avg(
-        cluster_counts_size_norm_path,
+    print("Computing the average value of each training column specified per cell meta cluster")
+    cell_meta_cluster_avgs = compute_cell_cluster_expr_avg(
+        cell_som_input_data_path,
         cell_som_cluster_cols,
         'cell_meta_cluster',
         keep_count=True
     )
 
     # save the average pixel SOM/meta counts per cell meta cluster
-    cell_meta_cluster_avgs_and_counts.to_csv(
-        os.path.join(base_dir, cell_meta_cluster_count_avg_name),
+    cell_meta_cluster_avgs.to_csv(
+        meta_expr_col_avg_path,
         index=False
     )
 
     print(
-        "Mapping meta cluster values onto average number of pixel SOM/meta cluster counts"
-        "across cell SOM clusters"
+        "Mapping meta cluster values onto average expression values across cell SOM clusters"
     )
 
     # read in the average number of pixel/SOM clusters across all cell SOM clusters
-    cell_som_cluster_avgs_and_counts = pd.read_csv(som_cluster_counts_avg_path)
+    cell_som_cluster_avgs = pd.read_csv(som_expr_col_avg_path)
 
     # merge metacluster assignments in
-    cell_som_cluster_avgs_and_counts = pd.merge_asof(
-        cell_som_cluster_avgs_and_counts, cell_cc.mapping, on='cell_som_cluster'
+    cell_som_cluster_avgs = pd.merge_asof(
+        cell_som_cluster_avgs, cell_cc.mapping, on='cell_som_cluster'
     )
 
     # resave average number of pixel/SOM clusters across all cell SOM clusters
     # with metacluster assignments
-    cell_som_cluster_avgs_and_counts.to_csv(
-        som_cluster_counts_avg_path,
+    cell_som_cluster_avgs.to_csv(
+        som_expr_col_avg_path,
         index=False
     )
 
-    # compute the weighted channel average expression per cell SOM cluster
+
+def generate_wc_avg_files(fovs, channels, base_dir, cell_cc,
+                          cell_som_input_data_name='cluster_counts_size_norm.feather',
+                          weighted_cell_channel_name='weighted_cell_channel.feather',
+                          cell_som_cluster_channel_avg_name='cell_som_cluster_channel_avg.csv',
+                          cell_meta_cluster_channel_avg_name='cell_meta_cluster_channel_avg.csv'):
+    """Generate the weighted channel average files per cell SOM and meta clusters.
```

Comment:
I think we can add a bit more text here, explaining when "weighted channel" makes sense. Something like "When using Pixie pixel clusters to do cell clustering, the number of each pixel cluster in each cell is tabulated. Here, we are multiplying these counts by the average expression profile of each pixel cluster to determine the weighted channel expression per cell."